src/main/java/com/azure/cosmos/kafka/connect/sink/StructToJsonMap.java

```diff
@@ -73,11 +73,48 @@ public static Map<String, Object> toJsonMap(Struct struct) {
                 case STRUCT:
                     jsonMap.put(fieldName, toJsonMap(struct.getStruct(fieldName)));
                     break;
+                case MAP:
+                    jsonMap.put(fieldName, handleMap(struct.getMap(fieldName)));
+                    break;
                 default:
                     jsonMap.put(fieldName, struct.get(fieldName));
                     break;
             }
         }
         return jsonMap;
     }
+
+    public static Map<String, Object> handleMap(Map<String, Object> map) {
+        if (map == null) {
+            return null;
+        }
+        Map<String, Object> cacheMap = new HashMap<>();
+        map.forEach((key, value) -> {
+            if (value instanceof Map) {
+                cacheMap.put(key, handleMap((Map<String, Object>) value));
+            } else if (value instanceof Struct) {
+                cacheMap.put(key, toJsonMap((Struct) value));
+            } else if (value instanceof List) {
+                List<Object> list = (List<Object>) value;
+                if (list.size() > 0 && list.get(0) instanceof Struct) {
```

Comment:
Is List expected to be of same types for all?

Reply:
Yes, list should have same types for all data types. However, I have updated the code to check for each item individually.